src/main/java/com/yahoo/bullet/bql/processor/ProcessedQuery.java

```diff
@@ -0,0 +1,259 @@
+
+    public QueryType getQueryType() {
+        return queryTypeSet.iterator().next();
+    }
+
+    public CountDistinctNode getCountDistinct() {
+        return countDistinctNodes.iterator().next();
+    }
+
+    public DistributionNode getDistribution() {
+        return distributionNodes.iterator().next();
+    }
+
+    public TopKNode getTopK() {
+        return topKNodes.iterator().next();
+    }
+
+    public Expression getExpression(ExpressionNode node) {
+        return expressionNodes.get(node);
+    }
+
+    public boolean hasAlias(ExpressionNode node) {
+        return aliases.containsKey(node);
+    }
+
+    public String getAlias(ExpressionNode node) {
+        return aliases.get(node);
+    }
+
+    public String getAliasOrName(ExpressionNode node) {
+        String alias = aliases.get(node);
+        return alias != null ? alias : node.getName();
+    }
+
+    public List<SelectItemNode> getNonAggregateSelectNodes() {
+        return selectNodes.stream().filter(node -> !isAggregate(node.getExpression())).collect(Collectors.toList());
+    }
+
+    public boolean isAggregateOrSuperAggregate(ExpressionNode node) {
+        return aggregateNodes.contains(node) || superAggregateNodes.contains(node);
+    }
+
+    public boolean isAggregate(ExpressionNode node) {
+        return aggregateNodes.contains(node);
+    }
+
+    public boolean isSuperAggregate(ExpressionNode node) {
+        return superAggregateNodes.contains(node);
+    }
+
+    public boolean isNotGroupByNode(ExpressionNode node) {
+        return !groupByNodes.contains(node);
+    }
+
+    public boolean isSimpleFieldExpression(ExpressionNode node) {
+        Expression expression = expressionNodes.get(node);
+        if (!(expression instanceof FieldExpression)) {
+            return false;
+        }
+        FieldExpression fieldExpression = (FieldExpression) expression;
+        return fieldExpression.getIndex() == null && fieldExpression.getKey() == null;
+    }
+
+    public boolean isSimpleAliasFieldExpression(ExpressionNode node) {
+        Expression expression = expressionNodes.get(node);
+        if (!(expression instanceof FieldExpression)) {
+            return false;
+        }
+        FieldExpression fieldExpression = (FieldExpression) expression;
+        return fieldExpression.getIndex() == null &&
+                fieldExpression.getKey() == null &&
+                aliases.values().contains(fieldExpression.getField());
+    }
+
+    public boolean isNotSimpleFieldExpression(ExpressionNode node) {
+        Expression expression = expressionNodes.get(node);
+        if (!(expression instanceof FieldExpression)) {
+            return true;
+        }
+        FieldExpression fieldExpression = (FieldExpression) expression;
+        return fieldExpression.getIndex() != null || fieldExpression.getKey() != null;
+    }
+
+    public boolean isNotSimpleAliasFieldExpression(ExpressionNode node) {
+        Expression expression = expressionNodes.get(node);
```

Comment:
Can't this just be !isSimpleAliasFieldExpression?

Reply:
Wait, you're a genius